proof_ipa.go

```diff
@@ -137,7 +141,7 @@ func MakeVerkleMultiProof(root VerkleNode, keys [][]byte, resolver NodeResolverF
 	return proof, pe.Cis, pe.Zis, pe.Yis, nil
 }
 
-func VerifyVerkleProof(proof *Proof, Cs []*Point, indices []uint8, ys []*Fr, tc *Config) bool {
+func VerifyVerkleProof(proof *Proof, Cs []*Point, indices []uint8, ys []*Fr, tc *Config) (bool, error) {
```

Comment:
A return of `false/true, nil` means that the proof verified or not verified cryptographically.
An error is more related to a formatting problem or some error not related to cryptography.

Distinguishing both things can be helpful to understand if the proof not passing might be because it is unsound, or it's because of some deserialization problem or incorrectly constructed.

@gballet, if you think that's not entirely useful and want to only return a `bool` or only a `error`; I'm OK with that and we can adjust a bit more here.